core/state_processor.go

```diff
@@ -99,6 +106,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		}
 		receipts = append(receipts, receipt)
 		allLogs = append(allLogs, receipt.Logs...)
+		if metrics.EnabledExpensive {
+			processTxTimer.UpdateSince(start)
+		}
```

Comment:
IMO we should call `processTxTimer.UpdateSince(start)` even if `applyTransaction()` fails.

Reply:
Fixed.